### PR TITLE
Add test for DECIMAL and CHAR pushdown in Postgres

### PR DESCRIPTION
**Pull request by kokosing**
Add test for DECIMAL and CHAR pushdown in Postgres

### CODE REVIEW

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java [hunk 1]
```diff
@@ -169,6 +169,40 @@ public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
         }
     }
 
+    @Test
+    public void testDecimalPredicatePushdown()
```
Comment (kokosing) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: test surely does not test what it says it tests. It checks if such query does not fail.

@findepi does it make sense to merge it? Or should we drop it?

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java [hunk 2]
```diff
@@ -169,6 +169,40 @@ public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
...
+                "(char_1 char(1), char_5 char(5), char_10 char(10))")) {
+            execute("INSERT INTO tpch.test_char_pushdown VALUES" +
+                    "('0', '0'    , '0'         )," +
+                    "('1', '12345', '1234567890')");
+
+            assertQuery("SELECT * FROM tpch.test_char_pushdown WHERE char_1 = '0' AND char_5 = '0    '",
+                    "VALUES ('0', '0    ', '0         ')");
+            assertQuery("SELECT * FROM tpch.test_char_pushdown WHERE char_5 = CHAR'12345' AND char_10 = '1234567890'",
+                    "VALUES ('1', '12345', '1234567890')");
+            assertQuery("SELECT * FROM tpch.test_char_pushdown WHERE char_10 = CHAR'0         '",
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: are the spaces still needed?

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java (hunk 1 above)
Comment (findepi) [reply presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: I think this is valuable, but be sure to add a TODO to make testing less brittle.

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java [hunk 3]
```diff
@@ -169,6 +169,40 @@ public void testInsertWithFailureDoesntLeaveBehindOrphanedTable()
...
+    public void testDecimalPredicatePushdown()
+            throws Exception
+    {
+        try (AutoCloseable ignoreTable = withTable("tpch.test_decimal_pushdown",
+                "(short_decimal decimal(9, 3), long_decimal decimal(30, 10))")) {
+            execute("INSERT INTO tpch.test_decimal_pushdown VALUES (123.321, 123456789.987654321)");
+
+            assertQuery("SELECT * FROM tpch.test_decimal_pushdown WHERE short_decimal <= 123.321",
+                    "VALUES (123.321, 123456789.987654321)");
+            assertQuery("SELECT * FROM tpch.test_decimal_pushdown WHERE long_decimal <= 123456789.987654321",
```
Comment (findepi) [inline presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: there should be a test with `=`, not only `<=`.
And one with `<=`  where value is bigger than the searched one.

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java (hunk 1 above)
Comment (kokosing) [reply presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: What do you mean with `to make testing less brittle`?

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java (hunk 2 above)
Comment (kokosing) [reply presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: no

File: presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java (hunk 1 above)
Comment (findepi) [reply presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: I mean to make the test in such a way that they not only check for correctness (as they try to do now), but also that the pushdown actually occurs (which is now assumed, not tested).

Comment (kokosing) [reply presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java]: This is what I also thought. However `less brittle` is very generic, so I thought you meant something more.

**Pull request closed without merging**